exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/ParquetReaderUtility.java

```diff
@@ -181,6 +185,71 @@ else if (parquetTableMetadata instanceof Metadata.ParquetTableMetadata_v2 &&
   }
 
   /**
+   * Checks that the metadata file was created by drill with version less than
+   * the version where was changed the serialization of BINARY values
+   * and assigns byte arrays to min/max values obtained from the deserialized string.
+   *
+   * @param parquetTableMetadata table metadata that should be corrected
+   */
+  public static void correctBinaryInMetadataCache(Metadata.ParquetTableMetadataBase parquetTableMetadata) {
+    if (hasOldBinarySerialization(parquetTableMetadata)) {
+      Set<List<String>> names = Sets.newHashSet();
+      if (parquetTableMetadata instanceof Metadata.ParquetTableMetadata_v2) {
```

Comment:
Added check for ParquetTableMetadata_v3, thanks. In class ColumnMetadata_v1 field `primitiveType` is serialized to the metadata cache file, so it is easier to check column type for this metadata version.